llvm/include/llvm/CodeGen/ValueTypes.td

```diff
@@ -101,244 +101,248 @@ def v256i1  : VTVec<256,  i1, 26>;  //  256 x i1 vector value
-
-def nxv1bf16  : VTScalableVec<1,  bf16, 176>;  // n x  1 x bf16 vector value
-def nxv2bf16  : VTScalableVec<2,  bf16, 177>;  // n x  2 x bf16 vector value
-def nxv4bf16  : VTScalableVec<4,  bf16, 178>;  // n x  4 x bf16 vector value
-def nxv8bf16  : VTScalableVec<8,  bf16, 179>;  // n x  8 x bf16 vector value
-def nxv16bf16 : VTScalableVec<16, bf16, 180>;  // n x 16 x bf16 vector value
-def nxv32bf16 : VTScalableVec<32, bf16, 181>;  // n x 32 x bf16 vector value
-
-def nxv1f32  : VTScalableVec<1,  f32, 182>;  // n x  1 x  f32 vector value
-def nxv2f32  : VTScalableVec<2,  f32, 183>;  // n x  2 x  f32 vector value
-def nxv4f32  : VTScalableVec<4,  f32, 184>;  // n x  4 x  f32 vector value
-def nxv8f32  : VTScalableVec<8,  f32, 185>;  // n x  8 x  f32 vector value
-def nxv16f32 : VTScalableVec<16, f32, 186>;  // n x 16 x  f32 vector value
-
-def nxv1f64  : VTScalableVec<1,  f64, 187>;  // n x  1 x  f64 vector value
-def nxv2f64  : VTScalableVec<2,  f64, 188>;  // n x  2 x  f64 vector value
-def nxv4f64  : VTScalableVec<4,  f64, 189>;  // n x  4 x  f64 vector value
-def nxv8f64  : VTScalableVec<8,  f64, 190>;  // n x  8 x  f64 vector value
+def v4096i1 : VTVec<4096, i1, 30>;  // 4096 x i1 vector value
+
+def v128i2  : VTVec<128,  i2, 31>;   //  128 x i2 vector value
+def v256i2  : VTVec<256,  i2, 32>;   //  256 x i2 vector value
+
+def v64i4   : VTVec<64,   i4, 33>;   //   64 x i4 vector value
+def v128i4  : VTVec<128,  i4, 34>;   //  128 x i4 vector value
+
+def v1i8    : VTVec<1,    i8, 35>;  //    1 x i8 vector value
+def v2i8    : VTVec<2,    i8, 36>;  //    2 x i8 vector value
+def v3i8    : VTVec<3,    i8, 37>;  //    3 x i8 vector value
+def v4i8    : VTVec<4,    i8, 38>;  //    4 x i8 vector value
+def v8i8    : VTVec<8,    i8, 39>;  //    8 x i8 vector value
+def v16i8   : VTVec<16,   i8, 40>;  //   16 x i8 vector value
+def v32i8   : VTVec<32,   i8, 41>;  //   32 x i8 vector value
+def v64i8   : VTVec<64,   i8, 42>;  //   64 x i8 vector value
+def v128i8  : VTVec<128,  i8, 43>;  //  128 x i8 vector value
+def v256i8  : VTVec<256,  i8, 44>;  //  256 x i8 vector value
+def v512i8  : VTVec<512,  i8, 45>;  //  512 x i8 vector value
+def v1024i8 : VTVec<1024, i8, 46>;  // 1024 x i8 vector value
+
+def v1i16    : VTVec<1,    i16, 47>;  //    1 x i16 vector value
+def v2i16    : VTVec<2,    i16, 48>;  //    2 x i16 vector value
+def v3i16    : VTVec<3,    i16, 49>;  //    3 x i16 vector value
+def v4i16    : VTVec<4,    i16, 50>;  //    4 x i16 vector value
+def v8i16    : VTVec<8,    i16, 51>;  //    8 x i16 vector value
+def v16i16   : VTVec<16,   i16, 52>;  //   16 x i16 vector value
+def v32i16   : VTVec<32,   i16, 53>;  //   32 x i16 vector value
+def v64i16   : VTVec<64,   i16, 54>;  //   64 x i16 vector value
+def v128i16  : VTVec<128,  i16, 55>;  //  128 x i16 vector value
+def v256i16  : VTVec<256,  i16, 56>;  //  256 x i16 vector value
+def v512i16  : VTVec<512,  i16, 57>;  //  512 x i16 vector value
```

Comment:
is there any reason to jump from 512 to 4096?

Reply:
Thank you for the review! The architecture we are working on only supports the larger value types; we do not need the shorter ones.

Reply:
right...I'm just wondering if you could add them as well if that's not too much of work.

Reply:
I did think about it, but decided to skip them, because I thought it might be a better practice to be conservative on reserving the indices and only reserve what will be used.